core/src/test/scala/unit/kafka/server/OffsetCommitRequestTest.scala

```diff
@@ -0,0 +1,147 @@
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.apache.kafka.common.requests.OffsetCommitRequestTest.assertResponseEquals
+import org.apache.kafka.common.{TopicPartition, Uuid}
+import org.apache.kafka.common.requests.{OffsetCommitRequest, OffsetCommitResponse}
+import org.apache.kafka.common.utils.Utils
+import org.junit.jupiter.api.{AfterEach, BeforeEach, Test, TestInfo}
+
+import java.util.Optional.empty
+import java.util.Properties
+import scala.collection.{Map, Seq}
+import scala.collection.immutable.ListMap
+import scala.jdk.CollectionConverters.{ListHasAsScala, MapHasAsJava, SeqHasAsJava}
+
+class OffsetCommitRequestTest extends BaseRequestTest {
+  override def brokerCount: Int = 1
+
+  val brokerId: Integer = 0
+  val offset = 15L
+  val groupId = "groupId"
+
+  var consumer: KafkaConsumer[_, _] = _
+
+  override def brokerPropertyOverrides(properties: Properties): Unit = {
+    properties.put(KafkaConfig.BrokerIdProp, brokerId.toString)
+    properties.put(KafkaConfig.OffsetsTopicPartitionsProp, "1")
+    properties.put(KafkaConfig.OffsetsTopicReplicationFactorProp, "1")
+  }
+
+  @BeforeEach
+  override def setUp(testInfo: TestInfo): Unit = {
+    super.setUp(testInfo)
+    consumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId)
+    consumer = createConsumer()
+  }
+
+  @AfterEach
+  override def tearDown(): Unit = {
+    if (consumer != null)
+      Utils.closeQuietly(consumer, "KafkaConsumer")
+    super.tearDown()
+  }
+
+  def createTopics(topicNames: String*): Seq[NameAndId] = {
+    topicNames.map(topic => {
+      createTopic(topic)
+      val topicId: Uuid = getTopicIds().get(topic) match {
+        case Some(x) => x
+        case _ => throw new AssertionError("Topic ID not found for " + topic)
+      }
+      NameAndId(topic, topicId)
+    })
+  }
+
+
+  @Test
+  def testTopicIdsArePopulatedInOffsetCommitResponses(): Unit = {
+    val topics = createTopics("topic1", "topic2", "topic3")
+    consumer.subscribe(topics.map(_.name).asJava)
+
+    sendOffsetCommitRequest(
+      topics.map((_, ListMap(0 -> offset))),
+      topics.map((_, Map(0 -> Errors.NONE))),
+      ApiKeys.OFFSET_COMMIT.allVersions().asScala
+    )
+  }
+
+  @Test
+  def testCommitOffsetFromConsumer(): Unit = {
+    val topics = createTopics("topic1", "topic2", "topic3")
+    consumer.commitSync(offsetsToCommit(topics, offset))
+  }
+
+  @Test
+  def testOffsetCommitWithUnknownTopicId(): Unit = {
+    val topics = createTopics("topic1", "topic2", "topic3")
+
+    sendOffsetCommitRequest(
+      topics.map((_, ListMap(0 -> offset))) ++ Seq((NameAndId("unresolvable"), ListMap(0 -> offset))),
+      Seq((NameAndId("unresolvable"), ListMap(0 -> Errors.UNKNOWN_TOPIC_ID))) ++ topics.map((_, Map(0 -> Errors.NONE))),
+      ApiKeys.OFFSET_COMMIT.allVersions().asScala.filter(_ >= 9)
+    )
+  }
+
+  @Test
+  def alterConsumerGroupOffsetsDoNotUseTopicIds(): Unit = {
```

Comment:
This test does not seem to be at the right place. It seems to me that `OffsetCommitRequestTest` is more focused on testing the OffsetCommitRequest API.

Reply:
Agreed, I thought to put it there because the underlying RPC is used, but you are right, it is a different client-level API.

Reply:
I removed the test since this method is already exercised in `org.apache.kafka.clients.admin.KafkaAdminClientTest#testOffsetCommitNumRetries`.